File: android/src/main/java/com/urbanairship/reactnative/CustomMessageActivity.java
/* Copyright 2017 Urban Airship and Contributors */

package com.urbanairship.reactnative;
import android.content.Intent;
import android.os.Bundle;

import com.urbanairship.messagecenter.MessageActivity;

public class CustomMessageActivity extends MessageActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState){
        super.onCreate(savedInstanceState);

        if (getIntent() != null && "CLOSE".equals(getIntent().getAction())){
            CustomMessageActivity.this.finish();
            return;
        }
    }

    @Override
    protected void onNewIntent(Intent intent){
        super.onNewIntent(intent);

        if (intent != null && "CLOSE".equals(intent.getAction())) {
            finish();
            return;
        }
    }

}

File: urbanairship-react-native/android/src/main/java/com/urbanairship/reactnative/CustomMessageActivity.kt
/* Copyright Urban Airship and Contributors */

package com.urbanairship.reactnative

import android.content.Intent
import android.os.Bundle
import com.urbanairship.messagecenter.MessageActivity

class CustomMessageActivity : MessageActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        intent?.let {
            if (UrbanAirshipReactModule.CLOSE_MESSAGE_CENTER == it.action) {
                finish()
            }
        }
    }

    override fun onNewIntent(intent: Intent) {
        super.onNewIntent(intent)

        if (UrbanAirshipReactModule.CLOSE_MESSAGE_CENTER == intent.action) {
            finish()
        }
    }
}